lib/router/src/lib.rs

```diff
@@ -105,37 +148,62 @@ impl<Req, Rec, Mk> Router<Req, Rec, Mk>
 where
     Rec: Recognize<Req>,
     Mk: Make<Rec::Target>,
-    Mk::Value: svc::Service<Req> + Clone,
+    <Mk as Make<Rec::Target>>::Value: Clone,
+    Mk::Value: svc::Service<Req>,
 {
-    pub fn new(recognize: Rec, make: Mk, capacity: usize, max_idle_age: Duration) -> Self {
-        Router {
-            inner: Arc::new(Inner {
+    pub fn new(
+        recognize: Rec,
+        make: Mk,
+        capacity: usize,
+        max_idle_age: Duration,
+    ) -> (Self, PurgeCache<Rec::Target, LoadShed<Mk::Value>>) {
+        let (cache, cache_bg) = Cache::new(capacity, max_idle_age);
+        let router = Self {
+            inner: Inner {
                 recognize,
                 make,
-                cache: Mutex::new(Cache::new(capacity, max_idle_age)),
-            }),
-        }
+                cache,
+            },
+        };
+
+        (router, cache_bg)
     }
 }
 
-impl<Req, Rec, Mk, Svc> svc::Service<Req> for Router<Req, Rec, Mk>
+impl<Req, Rec, Svc> Router<Req, Rec, FixedMake<Rec::Target, Svc>>
 where
     Rec: Recognize<Req>,
-    Mk: Make<Rec::Target, Value = Svc>,
     Svc: svc::Service<Req> + Clone,
-    Svc::Error: Into<error::Error>,
+{
+    pub fn new_fixed(recognize: Rec, routes: IndexMap<Rec::Target, Svc>) -> Self {
+        let capacity = routes.len();
+        let (router, _) = Self::new(
+            recognize,
+            FixedMake(routes),
+            capacity,
+            Duration::from_secs(1),
```

Comment:
Probably worth noting in a comment that the drop of the daemon task ensures that eviction is not performed and that the max_idle_age is ignored, etc.